src/core/api/link_metrics_api.cpp

```diff
@@ -49,12 +49,40 @@ otError otLinkMetricsQuery(otInstance *                aInstance,
                            otLinkMetricsReportCallback aCallback,
                            void *                      aCallbackContext)
 {
-    OT_ASSERT(aDestination != nullptr && aLinkMetricsFlags != nullptr);
+    OT_ASSERT(aDestination != nullptr);
 
     static_cast<Instance *>(aInstance)->Get<LinkMetrics>().SetLinkMetricsReportCallback(aCallback, aCallbackContext);
 
     return static_cast<Instance *>(aInstance)->Get<LinkMetrics>().LinkMetricsQuery(
-        static_cast<const Ip6::Address &>(*aDestination), aSeriesId, *aLinkMetricsFlags);
+        static_cast<const Ip6::Address &>(*aDestination), aSeriesId, aLinkMetricsFlags);
+}
+
+otError otLinkMetricsConfigForwardTrackingSeries(otInstance *                      aInstance,
+                                                 const otIp6Address *              aDestination,
+                                                 uint8_t                           aSeriesId,
+                                                 const otLinkMetricsSeriesFlags *  aSeriesFlags,
+                                                 const otLinkMetrics *             aLinkMetricsFlags,
+                                                 otLinkMetricsMgmtResponseCallback aCallback,
+                                                 void *                            aCallbackContext)
+{
+    OT_ASSERT(aDestination != nullptr && aSeriesFlags != nullptr && aLinkMetricsFlags != nullptr);
```

Comment:
`aLinkMetricsFlags` could be null if `*aSeriesFlags` is 0?

1.2 spec 4.11.3.4.4.4
> This field MUST NOT be included when Forward Series Flags='0'

The API design should conform with spec to avoid unnecessary confusion?